DetDescr/python/HcalGeometry.py

```diff
@@ -0,0 +1,106 @@
+    """
+
+    def __init__(self) :
+
+        # parameters that must align with the geometry
+        self.detectors_valid = [ ]
+        self.ThicknessScint = 0.
+        self.WidthScint = 0.
+        self.ZeroLayer = []
+        self.ZeroStrip = []
+        self.LayerThickness = []
+        self.NumLayers = []
+        self.NumStrips = []
+        self.HalfTotalWidth = []
+        self.NumSections = 0
+        self.verbose = 0
+
+    def __str__(self) :
+        """Stringify this configuration class"""
+
+        s = 'HcalReadoutGeometry { Scintillator thickness: %.1f mm, width: %.1f mm, and half total width (back Hcal): %.1f mm  \n' % ( 
+                self.ThicknessScint , self.WidthScint , self.HalfTotalWidthBack )
+        s += 'Z positions of zero-th layer %s mm, Z positions of zero-th strip %s mm \n'%(','.join(str(x) for x in self.ZeroLayer),','.join(str(x) for x in self.ZeroStrip))
+        s += 'layer thickness %s mm \n'%(','.join(str(x) for x in self.LayerThickness))
+        s += 'nlayers %s, nstrips %s }'%(' '.join(str(x) for x in self.NumLayers), ' '.join(str(x) for x in self.NumStrips))
+        return s
+
+class HcalGeometry() :
+    """Container for the various geometries
+
+    """
+
+    def __init__(self):
+        self.make_v12()
+    
+    def make_v12(self) :
+        """Create the HcalGeometry with the v12 geometry parameters
+
+        Only sets parameters that must align with the Hcal gdml constants.
+
+        - ThicknessScint: Scintillator thickness (in z).
+          @param gdml see: `scint_thick`.
+        
+        - WidthScint: Scintillator width (in x).
+          Can be obtained by: hcal_length/n_strips.
+
+        - ZeroLayer: Position of the first layer 
+          Back Hcal first layer (in z) starts after Ecal+Side-Hcal_z
+          Side Hcal first layer (in x or y) starts after Ecal_x/y / 2
+          For back Hcal: ecal_front_z + hcal_side_dz (in z)
+          For side Hcal: hcal_side_dz/2 (in x/y)
+          @param gdml see: `ecal_front_z` and `hcal_side_dz`
+
+        - ZeroStrip: Position of the first strip.
+          For back Hcal: back_transverse_width/2 (in x)
+          For side Hcal: ecal_front_z (in z)
+          @param gdml see: `hcal_back_dx` and `ecal_front_z`
+
+        - LayerThickness: 
+          Layer thickness (in z) 
+          Can be obtained by: absorber_thickness + scint_thickness + 2.0*air_thickness
+          @param gdml see: `side_abso_thick` and `back_abso2_thick` and `air_thick`
+
+        - NumLayers:
+          Number of layers per section.
+
+        - NumStrips: 
+          Number of strips per layer.
+
+        - NumSections:
+          Number of sections. 
+          Encoded in HcalID.
+          Back, Top, Left, Right, Bottom.
+
+        - HalfTotalWidth: 
+          For back Hcal: nstrips * scint_width / 2 (in x/y)
+          For side Hcal: (nlayers_otherside * layer_thick + Ecal_dx(y)/2)/2
+          @param gdml see: sideTB_dx, sideTB_dy
```

Comment:
Just want to say that this is :partying_face: awesome :tada: Great documentation!

Reply:
While exploring sideHcal pulse attenuation, I ended up adding Ecal dx/dy parameters and fixing a small bug on the halfwidth calculation for sideHcal: [here](https://github.com/LDMX-Software/ldmx-sw/blob/a7beaf9d9cc5a5222921dba686895006419b8437/DetDescr/python/HcalGeometry.py#L108-L114)